yarn-project/ethereum/src/types.ts

```diff
@@ -31,5 +31,5 @@ export type ViemClient = ViemPublicClient | ExtendedViemWalletClient;
 export type ViemContract<TAbi extends Abi> = GetContractReturnType<TAbi, ExtendedViemWalletClient>;
 
 export function isExtendedClient(client: ViemClient): client is ExtendedViemWalletClient {
-  return 'account' in client;
+  return 'account' in client && client.account !== undefined;
 }
```

Comment:
Is this method still in use after the refactor?

Reply:
yes, to strong type-check in L1TxUtils constructor and a couple more places (that could also be split into readOnly possibly)